python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+                visitIndex = self.extensionInfo.iloc[extensionIndex]['visitIndex']
+
+                self.sourceIndices[extensionIndex] = selected.selected
+
+                wcs = self.extensionInfo['wcs'][extensionIndex]
+                associations.reprojectWCS(wcs, fieldIndex)
+
+                associations.addCatalog(wcs, "STELLAR", visitIndex, fieldIndex,
+                                        instrumentIndex, detectorIndex, extensionIndex, isStar,
+                                        selected.sourceCat['x'].to_list(), selected.sourceCat['y'].to_list(),
+                                        np.arange(len(selected.sourceCat)))
+
+        associations.sortMatches(fieldIndex, minMatches=self.config.minMatches,
+                                 allowSelfMatches=self.config.allowSelfMatches)
+
+    def makeYAML(self, inputVisitSummary, inputFile=None):
+        """Make a YAML-type object that describes the parameters of the fit
+        model.
+
+        Parameters
+        ----------
+        inputVisitSummary : `list` of `lsst.afw.table.ExposureCatalog`
+            List of catalogs with per-detector summary information
+        inputFile : `str`
+            Path to a file that contains a basic model
+
+        Returns
+        -------
+        inputYAML : `wcsfit.YAMLCollector`
+            YAML object containing the model description
+        """
+        if inputFile is not None:
+            inputYAML = wcsfit.YAMLCollector(inputFile, "PixelMapCollection")
+        else:
+            inputYAML = wcsfit.YAMLCollector("", "PixelMapCollection")
+        inputDict = {}
+        modelComponents = ["INSTRUMENT/DEVICE", "EXPOSURE"]
+        baseMap = {"Type": "Composite", "Elements": modelComponents}
+        inputDict["EXPOSURE/DEVICE/base"] = baseMap
+
+        xMin = str(inputVisitSummary['bbox_min_x'].min())
+        xMax = str(inputVisitSummary['bbox_max_x'].max())
+        yMin = str(inputVisitSummary['bbox_min_y'].min())
+        yMax = str(inputVisitSummary['bbox_max_y'].max())
+
+        deviceModel = {"Type": "Composite", "Elements": self.config.deviceModel.list()}
+        inputDict["INSTRUMENT/DEVICE"] = deviceModel
+        for component in self.config.deviceModel:
+            if "poly" in component.lower():
+                componentDict = {"Type": "Poly",
+                                 "XPoly": {"OrderX": self.config.devicePolyOrder,
+                                           "SumOrder": True},
+                                 "YPoly": {"OrderX": self.config.devicePolyOrder,
+                                           "SumOrder": True},
+                                 "XMin": xMin, "XMax": xMax, "YMin": yMin, "YMax": yMax}
+            elif "identity" in component.lower():
+                componentDict = {"Type": "Identity"}
+            else:
+                raise ValueError(f"deviceModel component {component} is not supported")
```

Comment:
These are configs, so I think you should validate them as part of `config.validate()`, so that the failure happens on task init.